corehq/util/couch_doc_processor.py

```diff
@@ -147,6 +147,42 @@ def should_process(self, doc):
         return True
 
 
+class ProcessorProgressLogger(object):
+    def progress_starting(self, total, previously_visited):
+        print("Processing {} documents{}: ...".format(
```

Comment:
Since the generic iterator doesn't know about doc types anymore it's a bit hard to pass then to the logger.